src/formatDate.js

```diff
@@ -50,7 +50,41 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const initialDate = date.split(fromFormat[3]);
+  const convertedDate = new Array(3);
+
+  const initialDayIndex = fromFormat.indexOf('DD');
+  const initialMonthIndex = fromFormat.indexOf('MM');
+  const initialYearShort = fromFormat.indexOf('YY') !== -1;
+  const initialYearIndex = Math.max(
+    fromFormat.indexOf('YY'),
+    fromFormat.indexOf('YYYY')
+  );
+
+  const convertedDayIndex = toFormat.indexOf('DD');
+  const convertedMonthIndex = toFormat.indexOf('MM');
+  const convertedYearShort = toFormat.indexOf('YY') !== -1;
+  const convertedYearIndex = Math.max(
+    toFormat.indexOf('YY'),
+    toFormat.indexOf('YYYY')
+  );
+
+  convertedDate[convertedDayIndex] = initialDate[initialDayIndex];
+  convertedDate[convertedMonthIndex] = initialDate[initialMonthIndex];
```

Comment:
Instead of creating constants for every index, better refactor it to follow next logic:
Divide your logic into the next simple steps:
- split `from` and `to` arrays by separator;
- create an empty object to hold date parts;
- loop the `from` array and add props to that object. You need to have 'DD', 'MM', 'YY', and 'YYYY';
- create an empty formatted date array;
- loop the `to` array and push into it values from date object using `to[i]` values as keys;
- return joined array.